cranelift/codegen/src/binemit/memorysink.rs

```diff
@@ -78,13 +78,24 @@ pub trait RelocSink {
     fn reloc_block(&mut self, _: CodeOffset, _: Reloc, _: CodeOffset);
 
     /// Add a relocation referencing an external symbol at the current offset.
-    fn reloc_external(&mut self, _: CodeOffset, _: Reloc, _: &ExternalName, _: Addend);
+    fn reloc_external(
+        &mut self,
+        _: CodeOffset,
+        _: SourceLoc,
+        _: Reloc,
+        _: &ExternalName,
+        _: Addend,
+    );
 
     /// Add a relocation referencing a constant.
     fn reloc_constant(&mut self, _: CodeOffset, _: Reloc, _: ConstantOffset);
 
     /// Add a relocation referencing a jump table.
     fn reloc_jt(&mut self, _: CodeOffset, _: Reloc, _: JumpTable);
+
+    /// Track a call site which return address is the given CodeOffset, for the given opcode. Does
+    /// nothing in general, only useful for certain embedders (Spidermonkey).
+    fn add_call_site(&mut self, _: Opcode, _: CodeOffset, _: SourceLoc) {}
```

Comment:
I want to understand a bit more the need for this -- is this to allow for stack traces, in the absence of actual debug info? (Basically I'm looking ahead to cleaning up the diff w.r.t. mainline and wondering if this is a temporary or permanent/needed API change.)